examples/llama_recipes/README.md

```diff
@@ -0,0 +1,55 @@
+# Llama-recipes example
+This example demonstrates how to fine-tune and serve a Llama 2 model with llama-recipes for submission in the LLM efficiency challenge using the [toy-submission](../../toy-submission/) as a template.
+Llama-recipes provides an easy way to fine-tune a Llama 2 model with custom datasets using efficient techniques like LoRA or Llama-adapters.
+
+# Getting started
+In order to use llama-recipes we need to install the following pip package:
+```
+pip install llama-recipes
+```
+
+To obtain access to the model weights you need to fill out this [form](https://ai.meta.com/resources/models-and-libraries/llama-downloads/) to accept the license terms and acceptable use policy.
+
+After access has been granted, you need to acknowledge this in your HuggingFace account for the model you want to fine-tune. In this example we will continue with the 7B parameter version available under this identifier: meta-llama/Llama-2-7b-hf
+In this example the authentication will be done through a token created in the settings of your HuggingFace account.
+When running the Docker, the token will be expected in an environment variable that can be created with:
+
+```bash
+export  HUGGINGFACE_TOKEN="YOUR_HUGGINGFACE_TOKEN"
+```
+Make sure that the token allows read access to meta-llama/Llama-2-7b-hf.
+
+# Fine-tune the model
+Fine-tuning the model on one of the preconfigured datasets can than be done with a single line command line:
+```bash
+python -m llama_recipes.finetuning  --use_peft --peft_method lora --quantization --model_name meta-llama/Llama-2-7b-hf --output_dir peft_model_output
+```
+
+To use load a custom dataset we need to implement a function (get_custom_datset) following [this](https://github.com/facebookresearch/llama-recipes/blob/main/examples/custom_dataset.py) example.
```

Comment:
this custom dataset is a dialog, majority of eval datasets wont be dialog based so curious what's the specific interface people need to immplement here

Reply:
Yes, this is just demonstrating the interface to use a custom dataset, but you're not fixed to that selection. Will highlight that in a note.